pkg/cloudprovider/cloudprovider.go

```diff
@@ -223,16 +236,19 @@ func (c *CloudProvider) isAMIDrifted(ctx context.Context, machine *v1alpha5.Mach
 	if err != nil {
 		return false, fmt.Errorf("getting amis, %w", err)
 	}
-	// Get InstanceID to fetch from EC2
-	instanceID, err := utils.ParseInstanceID(machine.Status.ProviderID)
-	if err != nil {
-		return false, err
-	}
-	instance, err := c.instanceProvider.Get(ctx, instanceID)
+	return !lo.Contains(lo.Keys(amis), *instance.ImageId), nil
+}
+
+func (c *CloudProvider) isSubnetDrifted(ctx context.Context, instance *ec2.Instance,
```

Comment:
Are we getting to the point where the set of drift functions has expanded to the point that we should create a separate `drift.go` file for organization?

Reply:
Yeah, I'll let @engedaam handle this refactor in the following PRs if that's alright with you.